js/backend.js

```diff
@@ -47,81 +45,114 @@ $('#trainNew').on('click', function () {
-            if (!pyReturn) {
-                alert('Something went wrong');
-                return -1;
-            }
-        });
-    });
+    if (this.id === "trainAll") {
+      console.log("Annotating all documents...");
+      isAllDocuments = true;
+    }
+    else {
+      console.log("Annotating document: \"" + tagModel.currentDoc.title + "\"");
+      isAllDocuments = false;
+    }
+  }
+
+  // TODO: replace options
+  var options = {
+      args: ['--raw_data', tagModel.jsonifyData(isAllDocuments), '--n_iter', 30, '--model', tagModel.currentModel]
+  };
+  // try app
+  // TODO: replace with annotate all
+  let pyReturn;
+  // try installer path
+  launchPy(trainScript, options).then(function (data) {
+      pyReturn = data;
+      alert('!');
+      next();
+  }).catch(function () {
+      //try compiled path
+      launchPy(trainScriptLocal, options).then(function (data) {
+          pyReturn = data;
+          alert('!!!');
+          next();
+      }).catch(function () {
+          // still didn't work
+          if (!pyReturn) {
+              alert('Something went wrong');
+              return -1;
+          }
+      });
+  });
 });
 
-$('#annotateBtn').on('click', function () {
-    if (!modelPath) {
-        alert("Please add a model path");
-        return;
-    }
-    // replace options
-    var options = {
-        args: ['--model_path', modelPath, '--data_path', tagModel.exportAsString()]
-    };
-    // try app
-    // TODO: replace with annotate all
-    let pyReturn;
-    launchPy(annotateScript, options).then(function (data) {
-        pyReturn = data;
-        alert('!');
-        next();
-    }).catch(function () {
-        //try compiled path
-        launchPy(annotateScriptLocal, options).then(function (data) {
-            pyReturn = data;
-            alert('!!!');
-            next();
-        }).catch(function () {
-            // still didn't work
-            if (!pyReturn) {
-                alert('Something went wrong');
-                return -1;
-            }
-        });
-    });
+$('.annButton').on('click', function () {
+  var isAllDocuments;
 
-    next = function () {
-        console.log(pyReturn);
-    };
+  if (!tagModel.currentModel) {
+    alert("Please add a model first!");
+    return;
+  }
+  if (tagModel.currentDoc == null) {
+    alert("Please upload a document first!");
+    return
+  }
+
+  if (this.id === "annAll") {
+    console.log("Annotating all documents...");
+    isAllDocuments = true;
+  }
+  else if (this.id === "annThis"){
+    console.log("Annotating document: \"" + tagModel.currentDoc.title + "\"");
+    isAllDocuments = false;
+  }
+
+  // replace options
+  var options = {
+      args: ['--model', tagModel.currentModel, '--raw_data', tagModel.jsonifyData(isAllDocuments)]
+  };
+  // try app
+  // TODO: replace with annotate all
+  let pyReturn;
+  launchPy(annotateScript, options).then(function (data) {
+      pyReturn = data;
+      alert('!');
```

Comment:
This should probably be its own method, but this should get the results from the ml :
`                const fs = require('fs');
                let rawContent = fs.readFileSync('data.json');
                let jsonContent = JSON.parse(rawContent);
                console.log("ML returned \n", jsonContent);
                let errors = loadJsonData(jsonContent, "", obliterate = true);
                console.log('error displaying ml data: ', errors);`

Reply:
@tuj52136 Do you mind doing this commit? Not really sure what's going on here without formatting and don't want to break anything.